src/hooks/resources/tags.ts

```diff
@@ -0,0 +1,20 @@
+import * as _ from 'lodash';
+import { sbvrUtils } from '../../platform';
+import { checkTagKeyValidity } from '../../lib/tags';
+
+// Tag hooks
+const addTagHooks = (resource: string) => {
+	const nameProp = 'tag_key';
+	const tagHook: sbvrUtils.Hooks = {
+		POSTPARSE: ({ request }) => {
+			if (request.values[nameProp] != null) {
+				checkTagKeyValidity(request.values[nameProp]);
+			}
+		},
+	};
+
+	sbvrUtils.addPureHook('POST', 'resin', resource, tagHook);
+	sbvrUtils.addPureHook('PUT', 'resin', resource, tagHook);
```

Comment:
Ideally only the tag value should be updated, so we might want to also run the `tagHook` for PATCH requests as well in open-balena-api.